translator/internaldata/oc_to_traces.go

```diff
@@ -159,13 +157,20 @@ func ocSpanToInternal(src *octrace.Span, dest pdata.Span) {
 	ocSameProcessAsParentSpanToInternal(src.SameProcessAsParentSpan, dest)
 }
 
-func hasBytesValue(s []byte) bool {
-	for _, b := range s {
-		if b != 0 {
-			return true
-		}
-	}
-	return false
+// Transforms the byte slice trace ID into a [16]byte internal pdata.TraceID.
+// If larger input then it is truncated to 16 bytes.
+func traceIDToInternal(traceID []byte) pdata.TraceID {
```

Comment:
Perhaps it would be useful to have error handling for the cases when input byte slice is not the expected length. We should probably accept everything in the [0..16] range and reject anything longer. It likely "never" happens but who knows what we see in the input. Do you think it is fine to silently truncate?

Reply:
I think it is fine to silently truncate, we have it documented in both protos otlp and opencensus that the size should be 16.